crates/pallets/starknet/src/lib.rs

```diff
@@ -291,6 +307,115 @@ pub mod pallet {
             Ok(())
         }
 
+		// Submit a Starknet declare transaction.
+        ///
+        /// # Arguments
+        ///
+        /// * `origin` - The origin of the transaction.
+        /// * `transaction` - The Starknet transaction.
+        ///
+        ///  # Returns
+        ///
+        /// * `DispatchResult` - The result of the transaction.
+        ///
+        /// # TODO
+        /// * Compute weight
+        #[pallet::call_index(3)]
+        #[pallet::weight(0)]
+        pub fn add_declare_transaction(_origin: OriginFor<T>, transaction: Transaction) -> DispatchResult {
+            // TODO: add origin check when proxy pallet added
+
+            // Check if contract is deployed
+            ensure!(ContractClassHashes::<T>::contains_key(transaction.sender_address), Error::<T>::AccountNotDeployed);
+
+			// Check class hash is not already declared
+			ensure!(!ContractClasses::<T>::contains_key(transaction.call_entrypoint.class_hash.unwrap()), Error::<T>::ClassHashAlreadyDeclared);
+
+			// Check that contract class is not None
+			ensure!(transaction.contract_class.is_some(), Error::<T>::ContractClassMustBeSpecified);
+
+
+            let block = Self::current_block().unwrap();
+            let state = &mut Self::create_state_reader();
+			let contract_class = transaction.clone().contract_class.unwrap().to_starknet_contract_class();
```

Comment:
Instead of: 
![Screenshot 2023-03-27 at 09 28 05](https://user-images.githubusercontent.com/45264458/227870824-d157b218-1f06-4679-9a0e-99d2b4cfcdd3.png)

I would go with:
![Screenshot 2023-03-27 at 09 28 15](https://user-images.githubusercontent.com/45264458/227870871-985dbc3a-3943-447d-b85b-1be48acbe648.png)

to avoid unnecessary match arms.